### DISCUSSION TITLE
Can't install

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#501] [Everyday usage]
HEAD is now at 6968c199e Merge pull request #10284 from reitermarkus/reinstall-cask-options
fatal: Could not resolve HEAD to a revision
==> Installation successful!

why Could not resolve HEAD to a revision??

**Comment by gromgit** [comment]
What is the _full_ output of the install?

Also please post the output of `brew config` and `brew doctor`.

**Comment by marcelloromani** [comment]
Had the same error

OS: macOS Big Sur 11.2 (20D64)

Full ouput of the install:
```
$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
Password:
==> This script will install:
/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew

Press RETURN to continue or any other key to abort
==> /usr/bin/sudo /usr/sbin/chown -R marcelloromani:admin /usr/local/Homebrew
==> Downloading and installing Homebrew...
HEAD is now at 214e88aff Merge pull request #10541 from mistydemeo/fix_ohai_stdout_or_stderr
fatal: Could not resolve HEAD to a revision
==> Installation successful!

==> Homebrew has enabled anonymous aggregate formulae and cask analytics.
Read the analytics documentation (and how to opt-out) here:
  https://docs.brew.sh/Analytics
No analytics data has been sent yet (or will be during this `install` run).

==> Homebrew is run entirely by unpaid volunteers. Please consider donating:
  https://github.com/Homebrew/brew#donations

==> Next steps:
- Run `brew help` to get started
- Further documentation:
    https://docs.brew.sh
```

```
$ brew config
HOMEBREW_VERSION: 3.0.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ebc7cc36a0e029d58fac8cebd1b5ff583b3fd95b
Last commit: 12 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: HEAD
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 12.0 build 1200
Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4
```

```
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout master
```

**Comment by saintlammy** [comment]
I've got this same issue and trying to get it resolved as well.

**Comment by marcelloromani** [comment]
This is a blocker for me, I haven't been able to install _any_ package view `brew` since I reinstalled it on a fresh Big Sur MacPro.

Just tried to (re)install now and got:

```
From https://github.com/Homebrew/brew
 * [new branch]          dependabot/bundler/Library/Homebrew/sorbet-0.5.6277 -> origin/dependabot/bundler/Library/Homebrew/sorbet-0.5.6277
HEAD is now at 59dd425f4 Merge pull request #10579 from gibfahn/default_remote_name_tap
fatal: Could not resolve HEAD to a revision
Updated 1 tap (homebrew/cask).
```


If I try to install e.g. `wget` I get:

```
$ brew install wget
==> Searching for similarly named formulae...
Error: No similarly named formulae found.
Error: No available formula or cask with the name "wget".
==> Searching for a previously deleted formula (in the last month)...
Error: No previously deleted formula found.
==> Searching taps on GitHub...
Error: No formulae found in taps.
```

Surely this must be affecting quite a lot of people?

**Reply by Daisyweb** [reply]
Yes i have the same issue with Big Sur! What should I do?

**Reply by lexcraw4d** [reply]
same issue? Any resolutions?

**Reply by martinkiperszmidmeli** [reply]
Same issue

**Reply by gromgit** [reply]
What's the output of `brew doctor`?

**Reply by lagshetti20** [reply]
same issue for brew install gradle

**Reply by lagshetti20** [reply]
please suggest a solution for it

**Reply by gromgit** [reply]
What's the output of `brew doctor`?

**Comment by marcelloromani** [comment]
OK, so I was able to fix it like this:

```
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout master
```

I executed the command suggested by `doctor`:

```
$  git -C $(brew --repo homebrew/core) checkout master
Updating files: 100% (5765/5765), done.
Branch 'master' set up to track remote branch 'master' from 'origin'.
Already on 'master'
```

after that, the system seemed fine:

```
$ brew doctor
Your system is ready to brew.
```

Test: `brew install wget` would fail, but after the above steps it worked fine.

**Reply by wwwenjie** [reply]
It's works for me. Save my day! I thought there may network issue but actually not. Maybe it only appear in Big Sur.

**Reply by Daisyweb** [reply]
OMG ! IT works! Thank you! For the hint with doctor !

**Reply by ivoaxi** [reply]
It worked for me too on Big Sur. Great tip.

**Reply by viajeraune** [reply]
worked for me as well! Thank you so much!

**Reply by sonalimeh** [reply]
worked for me too. Thanks man.

**Reply by atxy2k** [reply]
Worked for me too in Big Sur. Thanks!

**Reply by daleharris541** [reply]
Couldn't figure out what was wrong until I found this. Thanks for the help! This fixed Big Sur with M1 Macbook Air (because why wouldn't you want to develop on the couch for 20 hours straight?)

**Reply by timothyokooboh** [reply]
Thanks a lot. It worked for me

**Reply by JasonChen233** [reply]
Thanks a a lot!!! The problem has been solved!!!

**Reply by IAMebonyhope** [reply]
Thanks a lot, been trying to resolve this for days.

**Reply by ghost** [reply]
Thanks a lot.

**Reply by AdityaLambdaTest** [reply]
Thank you so so much. It helped..

**Reply by riyol** [reply]
Thank you so much.

**Reply by nina000Chen** [reply]
非常感谢！

**Reply by Sarps** [reply]
Thanks a lot, this worked

**Reply by swayam-sixt** [reply]
rm -rf $(brew --repo homebrew/core)
brew tap homebrew/core  

The above two command works as well.

**Reply by vsc212** [reply]
This was a lifesaver ! Worked for me too. Thanks for sharing.

**Reply by EricXiongLi** [reply]
Useful, thanks a lot

**Reply by QT4124** [reply]
thank you

**Reply by Eugenoo** [reply]
I've had a problem with php installation on MacOS Monterey. After this commands all works fine :)

**Reply by aeestrada21** [reply]
Thank you this really help me notice what the problem was, i follow the steps after the brew doctor command and it solve the problem.

**Reply by Cleevemadjouh** [reply]
Could help me to install Homebrew on my MacBook pro M1  monterey ?

I got this issue:


MacBook-Pro ~ % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /usr/sbin/chown -R rone:admin /opt/homebrew
==> Downloading and installing Homebrew...
HEAD is now at c1a9ce62a Merge pull request #13163 from Homebrew/dependabot/bundler/Library/Homebrew/rubocop-rspec-2.10.0

**Comment by saintlammy** [comment]
I got it resolved this way on my BigSur also

> OK, so I was able to fix it like this:
> 
> 
> 
> ```
> 
> $ brew doctor
> 
> Please note that these warnings are just used to help the Homebrew maintainers
> 
> with debugging if you file an issue. If everything you use Homebrew for is
> 
> working fine: please don't worry or file an issue; just ignore this. Thanks!
> 
> 
> 
> Warning: Some taps are not on the default git origin branch and may not receive
> 
> updates. If this is a surprise to you, check out the default branch with:
> 
>   git -C $(brew --repo homebrew/core) checkout master
> 
> ```
> 
> 
> 
> I executed the command suggested by `doctor`:
> 
> 
> 
> ```
> 
> $  git -C $(brew --repo homebrew/core) checkout master
> 
> Updating files: 100% (5765/5765), done.
> 
> Branch 'master' set up to track remote branch 'master' from 'origin'.
> 
> Already on 'master'
> 
> ```
> 
> 
> 
> after that, the system seemed fine:
> 
> 
> 
> ```
> 
> $ brew doctor
> 
> Your system is ready to brew.
> 
> ```
> 
> 
> 
> Test: `brew install wget` would fail, but after the above steps it worked fine.

**Comment by fredngo** [comment]
This appears to still be an issue on May 25, 2021 on newly-installed homebrew on Big Sur -- not too difficult for experienced folks to fix but extremely confounding to beginners who are following installation guides. Any way to automate the running of `git -C $(brew --repo homebrew/core) checkout master` instead of forcing folks to run that on the Terminal?

**Reply by carlocab** [reply]
The difficulty with automating it is that it's not clear when exactly the command should be run for the user. There are a number of reasons why someone might have a non-master branch checked out, and I don't know of a way to ascertain whether they apply.

**Reply by gromgit** [reply]
I can think of only one case where an automated fix could be safely done: If `$(brew --repo homebrew/core)` doesn't contain `.git/config`, the initial clone was never completed, and so it should be safe to automatically:
```
rm -fr $(brew --repo homebrew/core)
brew tap homebrew/core
```

**Reply by fredngo** [reply]
As far as I can tell, it has happened only on Big Sur during Homebrew initial installation, if that helps at all.

I would suggest the checkout master of `homebrew/core` be automated only during initial installation. So there must be a way to do the same detection as `brew doctor` does during initial installation, and doing the checkout then -- as I think having `master` checked out during installation would be the expected state, right? This avoids any "number of reasons why someone might have a non-master branch checked out".

**Reply by bradtchapman** [reply]
Same experience here.  Reproducible on Big Sur 11.4 with an Intel Mac, Xcode Command Line Tools 12.5.  

I had to run `brew doctor` to get this note:
```
Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout master
```
After running the suggested command, it's fine now.

```
$ git -C $(brew --repo homebrew/core) checkout master
Updating files: 100% (5935/5935), done.
$ brew doctor
Your system is ready to brew.
$ brew search ffmpeg
==> Formulae
ffmpeg               ffmpeg2theora        ffmpeg@2.8           ffmpegthumbnailer
```

This is 100% reproducible removing and reinstalling brew.

edit: questions...

1. what is it about Big Sur that requires this extra step?  
2. is it possible to include this statement in the main branch and not impact macOS 10.15 and below?

**Reply by kristianrl** [reply]
I had the same issue on a fresh install of MacOS Monterey: @gromgit's solution solved the problem:

> ```
> rm -fr $(brew --repo homebrew/core)
> brew tap homebrew/core
> ```

**Comment by zmatin-sans** [comment]
ok so what do i do after i type in 
brew --repo homebrew/core

do i then reinstall homebrew? becuase i tried that and it still gave me the fatal: Could not resolve HEAD to a revision, because that didnt work either.
